tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -3017,3 +3018,170 @@ def call(self, inputs, state):
+               state_is_tuple=True, activation=None, reuse=None, name=None):
+    """Initialize the basic NLSTM cell.
+
+    Args:
+      num_units: int, The number of hidden units of each cell state
+        and hidden state.
+      depth: int, The number of layers in the nest
+      forget_bias: float, The bias added to forget gates.
+      state_is_tuple: If True, accepted and returned states are tuples of
+        the `h_state` and `c_state`s.  If False, they are concatenated
+        along the column axis.  The latter behavior will soon be deprecated.
+      activation: Activation function of the inner states.  Default: `tanh`.
+      reuse: (optional) Python boolean describing whether to reuse variables
+        in an existing scope.  If not `True`, and the existing scope already has
+        the given variables, an error is raised.
+      name: String, the name of the layer. Layers with the same name will
+        share weights, but to avoid mistakes we require reuse=True in such
+        cases.
+    """
+    super(NLSTMCell, self).__init__(_reuse=reuse, name=name)
+    if not state_is_tuple:
+      logging.warn("%s: Using a concatenated state is slower and will soon be "
+                   "deprecated.  Use state_is_tuple=True.", self)
+
+    # Inputs must be 2-dimensional.
+    self.input_spec = base_layer.InputSpec(ndim=2)
+    self._num_units = num_units
+    self._forget_bias = forget_bias
+    self._state_is_tuple = state_is_tuple
+    self._depth = depth
+    self._activation = activation or math_ops.tanh
+    self._kernels = None
+    self._biases = None
+    self.built = False
+
+  @property
+  def state_size(self):
+    if self._state_is_tuple:
+      return tuple([self._num_units] * (self.depth + 1))
+    else:
+      return self._num_units * (self.depth + 1)
+
+  @property
+  def output_size(self):
+    return self._num_units
+
+  @property
+  def depth(self):
+    return self._depth
+
+  def build(self, inputs_shape):
+    if inputs_shape[1].value is None:
+      raise ValueError("Expected inputs.shape[-1] to be known, saw shape: %s"
+                       % inputs_shape)
+
+    input_depth = inputs_shape[1].value
+    h_depth = self._num_units
+    self._kernels = []
+    self._biases = []
+    for i in range(self.depth):
+      if i == 0:
+        self._kernels.append(
+          self.add_variable(
+            "kernel_{}".format(i),
+            shape=[input_depth + h_depth, 4 * self._num_units]))
+      else:
+        self._kernels.append(
+          self.add_variable(
+            "kernel_{}".format(i),
+            shape=[2 * h_depth, 4 * self._num_units]))
+      self._biases.append(
+        self.add_variable(
+          "bias_{}".format(i),
+          shape=[4 * self._num_units],
+          initializer=init_ops.zeros_initializer(dtype=self.dtype)))
+
+    self.built = True
+
+  def recurrence(self, inputs, hidden_state, cell_states, depth):
+    """use recurrence to traverse the nested structure
```

Comment:
Nit: comment statements should end in a period.
https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments

Same below.